pkg/rancher-components/README.md

```diff
@@ -9,15 +9,78 @@ yarn
+
+1. Is this component used in multiple projects? For example, Rancher Dashboard and Rancher Desktop maintained individual implementations of checkbox, radio button, and input components. This duplication of effort was eliminated by including these components in Rancher Components.
+2. Is this component a core building block? Core building blocks are components like buttons, inputs, labels, badges, etc.. components used to build larger and more complicated features. We can think of core building blocks akin to atoms and molecules if you subscribe to the concept of [atomic design](https://bradfrost.com/blog/post/atomic-web-design/#atoms).
+3. Is this component self-contained? A component should be able to manage its state without relying on Vuex or any API calls.
+
+### Governance
+
+Governance is the most important ingredient to creating a healthy component library that stands the test of time. We must have clear intent and understanding of usage before adding a new component to Rancher Components. Existing Dashboard components might need review before porting so that we can better define clear boundaries between Dashboard and component behavior.
+
+![Rancher Components Docs 2022-11-28 09 19 05 excalidraw](https://user-images.githubusercontent.com/835961/204343084-5b1ad6d3-b9ca-4295-a81f-3f7a02de63f6.png)
+
+## Developing and debugging
+
+Rancher Dashboard is a monorepo that houses development for several packages, including Rancher Components. We use [Yarn Workspaces](https://classic.yarnpkg.com/lang/en/docs/workspaces/) to make developing in a monorepo a little easier.
+
+Running `yarn install` will automatically link inter-package dependencies for each package that has workspaces enabled. The only requirement for linking is that the version is the same in the linked package and the target library. For example, we want to make sure that the `@rancher/components` version matches the dependency for `@rancher/shell` to automatically link
+
+### `@rancher/components package.json`
+
 ```
-yarn lint
+{
+  "name": "@rancher/components",
+  "repository": "git://github.com:rancher/dashboard.git",
+  "license": "Apache-2.0",
+  "version": "0.1.0", // The version of @rancher/components that we want to link
+  ...
+}
+
+```
+
+### `@rancher/shell package.json`
+
 ```
+{
+  "name": "@rancher/shell",
+  "version": "0.1.4",
+  "description": "Rancher Dashboard Shell",
+  "repository": "https://github.com/rancherlabs/dashboard",
+  ...
+  "dependencies": {
+    ...
+    "@nuxtjs/webpack-profile": "0.1.0",
+    "@popperjs/core": "2.4.4",
+    "@rancher/components": "0.1.0", // @rancher/components version matches what is in package.json and will automatically link for development
+    ...
+  } 
+}
+
+```
+
+You can build and watch for changes in `@rancher/components` after ensuring that all `@rancher/components` dependencies match the version reflected in `package.json`
+
+```
+$ yarn build:lib --watch
```

Comment:
To confirm, once https://github.com/rancher/dashboard/pull/7425 merges, this line (excluding initial install and dep version alignment) is all that's needed when developing files in rancher-component and dashboard/shell at the same time?

Reply:
Correct. Making changes to Rancher Components and testing locally in dashboard might look something like 

```
$ yarn build:lib --watch
```

and in a separate terminal

```
$ API=https://rancher-backend ROUTER_BASE=/dashboard yarn dev
```

Alternatively, we could consider adding a new dev dependency, like [concurrently](https://www.npmjs.com/package/concurrently), to distill this down into a single script.

Reply:
I would be in favor of a single script without too much troubles, if was not a matter of resource. To be fair, as we are using Yarn modules, I wonder if there's not a way to serve (not build) everything as we did before, without run it twice. Probably a Webpack configuration or something.
In a first iteration I would rather opt for `concurrently` lib, so we do not have to change the workflow.

Reply:
It also does not make sense, if the plan is to finally change repository.